src/delagent/agent/delagent.c

```diff
@@ -157,7 +157,7 @@ int main (int argc, char *argv[])
     DeleteUpload(DelUpload); 
     fprintf(stdout, "The upload '%ld' is deleted by the user '%s'.\n", DelUpload, user_name);
   }
-  if (DelFolder) { DeleteFolder(DelFolder); }
+  if (DelFolder) { DeleteFolder(DelFolder, user_id); }
```

Comment:
If we encapsulate a function for folder check, then also userid is required for ListFoldersRecurse and deletefolder should pass the userid. BTW user permission check for folders are done in ListFoldersRecurse only.